src/main/java/com/sahil/SplitwiseApp/controller/SettlementController.java

```diff
@@ -0,0 +1,87 @@
+package com.sahil.SplitwiseApp.controller;
+
+import com.sahil.SplitwiseApp.DTO.DebtUsersDTO;
+import com.sahil.SplitwiseApp.DTO.PaymentDTO;
+import com.sahil.SplitwiseApp.DTO.SettlementDTO;
+import com.sahil.SplitwiseApp.constants.ApiConstants;
+import com.sahil.SplitwiseApp.model.Expenses;
+import com.sahil.SplitwiseApp.service.SettlementService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping(ApiConstants.BASE_URL)
+public class SettlementController {
+
+    @Autowired
+    SettlementService service;
+
+    @Autowired
+    DebtUsersController debtUsersController;
+
+    @Autowired
+    ExpensesController expensesController;
+
+    @Autowired
+    UsersController usersController;
+
+    @GetMapping("/users/{user1-Id}/settle/{user2-Id}")
+    public SettlementDTO debtUserToLenderHistory(@PathVariable("user1-Id") int debtUserId, @PathVariable("user2-Id") int lenderId) {
+        return service.debtUserToLenderHistory(debtUserId,lenderId);
+    }
+
+    @PutMapping("/users/{user1-Id}/settle/{user2-Id}")
+    public void settleDebtsBetweenUsers(@PathVariable("user1-Id") int debtUserId, @PathVariable("user2-Id") int lenderId){
+        service.settleDebtsBetweenUsers(debtUserId,lenderId);
+    }
+
+    @GetMapping("/users/{user-id}/to-pay")
```

Comment:
try naming this url, differently follow rest semantics.
where resource and action performed is considered